e2e/support/github/run-e2e-docker-env.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash -eu
+
+# get the dir containing the script
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+# create a temporary working directory
+working_dir=$(mktemp -d "${TMPDIR:-/tmp/}openmrs-e2e-frontends.XXXXXXXXXX")
+# get a list of all the apps in this workspace
+apps=$(yarn workspaces list --json | jq -r 'select((.location != ".") and (.location | test("form-entry-app") | not) and (.location | test("-app"))) | .name')
+# this array will hold all of the packed app names
+app_names=()
+
+echo "Creating packed archives of apps..."
+# for each app
+for app in $apps
+do
+  # @openmrs/esm-whatever -> _openmrs_esm_whatever
+  app_name=$(echo "$app" | tr '[:punct:]' '_');
+  # add to our array
+  app_names+=("$app_name.tgz");
+  # run yarn pack for our app and add it to the working directory
+  yarn workspace "$app" pack -o "$working_dir/$app_name.tgz" >/dev/null;
+done;
+echo "Created packed app archives"
+
+echo "Creating dynamic spa-assemble-config.json..."
+# dynamically assemble our list of frontend modules, prepending the login app and
+# primary navigation apps; apps will all be in the /app directory of the Docker
+# container
+jq -n \
+  --arg apps "$apps" \
+  --arg app_names "$(echo ${app_names[@]})" \
+  '{"@openmrs/esm-primary-navigation-app": "next", "@openmrs/esm-dispensing-app": "next", "@openmrs/esm-home-app": "next"} + (
```

Comment:
So, fundamentally, adding esm-dispensing-app here won't work. The problem is this will always refer to the _last_ version uploaded to NPM, not the current version. So the e2e tests won't test the code in the PR, they'll test the code in the most recently merged PR.

Reply:
Basically, we need to adjust the above so that it generates a `$working_dir/_openmrs_esm_dispensing_app.tgz` and then adds an entry like `"@openmrs/esm-dispensing-app": "/app/_openmrs_esm_dispensing_app.tgz"`.

Reply:
I think this can be done by switching the `jq` command above to:

```
jq -r 'select(.name | test("-app")) | .name'
```

And then switching the line:

```
yarn workspace "$app" pack -o "$working_dir/$app_name.tgz" >/dev/null;
```

To:

```
yarn pack -o "working_dir/$app_name.tgz" >/dev/null
```